src/main/java/frc/robot/subsystems/elevator/Elevator.java

```diff
@@ -0,0 +1,59 @@
+package frc.robot.subsystems.elevator;
+
+import static frc.robot.subsystems.elevator.ElevatorConstants.*;
+
+import com.ctre.phoenix6.signals.ControlModeValue;
+import edu.wpi.first.units.Distance;
+import edu.wpi.first.units.MutableMeasure;
+import edu.wpi.first.wpilibj.smartdashboard.Mechanism2d;
+import edu.wpi.first.wpilibj.smartdashboard.MechanismRoot2d;
+import edu.wpi.first.wpilibj2.command.Command;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import org.littletonrobotics.junction.Logger;
+
+public class Elevator extends SubsystemBase {
+
+    private static Elevator INSTANCE;
+    private final ElevatorInputs inputs = new ElevatorInputs();
+    private final ElevatorIO io;
+    private final Mechanism2d mechanism2d = new Mechanism2d(MECHANISM_WIDTH, MECHANISM_HEIGHT);
+    private final MechanismRoot2d root = mechanism2d.getRoot("Elevator", ROOT_X, ROOT_Y);
+    private ControlModeValue controlMode;
+    private Command lastCommand = null;
+    private Command currentCommand = null;
+    private boolean changedToDefaultCommand = false;
+
+    private Elevator(ElevatorIO io) {
+        this.io = io;
+    }
+
+    public void setPower(double power) {
+        inputs.power = power;
+        inputs.controlMode = ElevatorIO.ControlMode.PERCENT_OUTPUT;
+    }
+
+
+    public void setHeight(MutableMeasure<Distance> height) {
+        inputs.height = height;
+        inputs.controlMode = ElevatorIO.ControlMode.POSITION;
+    }
```

Comment:
Where do actual commands go to the IO?

Reply:
9921811df8729c75faef804d422cc84184d58cfa

Reply:
This is good but did you delete the inputs update? Also the name should be `currentHeight`/`setpointHeight` because just "height" is confusing.

Reply:
23dec654c9f9bdbecb9b8bcf18041cc462c6c19a

Reply:
This is good, however, why did you delete the control mode updates?